src/Features/CSharp/Portable/AddStaticModifier/CSharpAddStaticModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using Microsoft.CodeAnalysis.AddStaticModifier;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.AddStaticModifier
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(CSharpAddStaticModifierCodeFixProvider)), Shared]
+    internal sealed class CSharpAddStaticModifierCodeFixProvider : AbstractAddStaticModifierCodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(
+                "CS0708" // 'MyMethod': cannot declare instance members in a static class
+            );
+
+        [ImportingConstructor]
+        public CSharpAddStaticModifierCodeFixProvider()
+        {
+        }
+
+        protected override SyntaxNode MapToDeclarator(SyntaxNode node)
+        {
+            return node switch
```

Comment:
Is it fine to use the new syntax? Also I have noticed that a couple of analyzers are defining this method, do you want me to extract them into some helper?

Reply:
this is fine for now IMO.